### PR TITLE
adds mkdir

### PR DESCRIPTION
**Pull request by lgvaz**
Adds `mkdir` function as described in discord [here](https://discord.com/channels/689892369998676007/689892370002870284/993544265093357749).

I feel like my tests might be a bit convoluted? Please feel free to suggest changes :)

**NOTE:** The only file I really modified was `03_xtras.ipynb`, I guess all the other files were changed by nbdev hooks?

### CODE REVIEW

**Comment by lgvaz** [comment]
Hmm, some permission error on the CI

**Comment by jph00** [comment]
Yeah you can't delete a directory you're using in Windows. And your `with` block means you're using it.

Change your code so you don't use `with`, but instead use a `finally` block to clean up.

(PS: be sure to at-mention me on PRs so I see them)

**Comment by lgvaz** [comment]
Heh @jph00, I don't know how to do this 😅

**Comment by jph00** [comment]
Hey no worries. You're still using a `with` block for the directory. Your commit only changed the file.

**Comment by jph00** [comment]
Feel free to close the PR if you don't feel like debugging this -- it's not a big deal either way :)

**Comment by lgvaz** [comment]
No no, don't get me wrong, I really wanna do this, I just had spent and hour trying to understand the error and was not getting nowhere so I asked for help, but in hindsight "I don't know how to do this" is one of the worse ways for asking help, sorry about that.

Anyways, the directory is using `with`!!! I completely overlooked that and tunel visioned on the file, I'll try it again now :)

**Comment by lgvaz** [comment]
Yeeeee @jph00 had to be a bit hacky but it worked! I had to manually close the file and the directory for windows to be happy about deleting it hahahah. Let me know if you think there's a better way to do this, maybe again I just focused too much into using the temporary dir/files and missed an simpler approach.

Btw, should I tag you in every update in the PR or just once at the beggining?

**Comment by jph00** [comment]
This is using a private method, which isn't a good idea.

I think you're making this over complicated frankly. There are examples on stack overflow that'll show you how to get a temporary file name without actually creating anything. Afaict you should be fine to use "with" for the file, just not for the directory.

You only need to tag me at the start of a PR

**Comment by lgvaz** [comment]
Hi! Just want to say I have not forgotten about this. I just got my hands on a windows laptop so that will make it much easier to figure this out!

**Comment by lgvaz** [comment]
Hi Jeremy! It's been a while but I finally figured it out. As you suggested, I was overcomplicating: I don't need to create a temporary file inside a temporary directory, I can just create a normal file!!

The CI on windows is currently disabled, but I tested this on a windows machine and it works!

File: nbs/.gitattributes [hunk 1]
```diff
@@ -1 +1,2 @@
-*.ipynb merge=nbdev-merge
```
Comment (jph00) [inline nbs/.gitattributes:1]: Can you revert your change to this file please? Looks like you've got an old version of nbdev hooks installed that's created this change.

File: nbs/03_xtras.ipynb [hunk 2]
```diff
@@ -318,6 +318,40 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#|export\n",
+    "def mkdir(path, exist_ok=False, parents=False, overwrite=False, **kwargs):\n",
+    "    \"Creates and returns a directory defined by `path`, optionally removing previous existing directory if `overwrite` is `True`\"\n",
+    "    path = Path(path)\n",
```
Comment (jph00) [inline nbs/03_xtras.ipynb:330]: You should add `import shutil` here. Currently this import is only in a non-exported cell, so this won't work outside this notebook.

**Review by jph00 — CHANGES_REQUESTED** [review]
Looking very close now!

File: nbs/.gitattributes (hunk 1 above)
Comment (lgvaz) [reply nbs/.gitattributes]: done :)

File: nbs/03_xtras.ipynb (hunk 2 above)
Comment (lgvaz) [reply nbs/03_xtras.ipynb:331]: done! it should be inside the function right?

**Comment by lgvaz** [comment]
I see there's a conflict now. Is it because the cell number of my added function has changed compared to master?

**Comment by lgvaz** [comment]
Done! :)

**Comment by jph00** [comment]
Good stuff!

**Pull request merged**